### PR TITLE
Address fourth part of 477: Moving Entity.delete to Key (for datastore).

### PR DESCRIPTION
**Pull request by dhermes**
**NOTE**: Has #480 as diffbase.

### CODE REVIEW

**Comment by dhermes** [comment]
@tseaver rebased and ready to review here

File: gcloud/datastore/test_dataset.py [hunk 1]
```diff
@@ -157,6 +157,11 @@ def allocate_ids(self, dataset_id, key_pbs):
         num_pbs = len(key_pbs)
         return [_KeyProto(i) for i in range(num_pbs)]
 
+    def delete_entities(self, dataset_id, key_pbs):
```
Comment (tseaver) [inline gcloud/datastore/test_dataset.py]: Why is this method here?  Wouldn't it be easier / clearer to define a `_Connection` class in `test_key` which had only the methods we need there?

**Comment by tseaver** [comment]
LGTM, aside from borrowing a test fixture between testcase modules (I see that we were doing it already, so not really a blocker).

File: gcloud/datastore/test_dataset.py (hunk 1 above)
Comment (dhermes) [inline gcloud/datastore/test_dataset.py]: I was just trying to bail water out of the boat with these tests.

This class gets moved in #483

**Pull request merged**